R/variable_types.R

```diff
@@ -30,76 +30,23 @@
 #'   )
 #' )
 variable_types <- function(data, columns = NULL) {
-  UseMethod("variable_types")
-}
-
-#' @export
-variable_types.default <- function(data, columns = NULL) {
-  checkmate::assert_character(columns, null.ok = TRUE, any.missing = FALSE)
-
-  res <- if (is.null(columns)) {
-    vapply(
-      data,
-      function(x) class(x)[[1]],
-      character(1),
-      USE.NAMES = FALSE
-    )
-  } else if (checkmate::test_character(columns, any.missing = FALSE)) {
-    stopifnot(all(columns %in% names(data) | vapply(columns, identical, logical(1L), "")))
-    vapply(
-      columns,
-      function(x) ifelse(x == "", "", class(data[[x]])[[1]]),
-      character(1),
-      USE.NAMES = FALSE
-    )
-  } else {
-    character(0)
-  }
-
-  return(res)
-}
-
-#' @export
-variable_types.data.frame <- function(data, columns = NULL) { # nolint: object_name_linter.
-  variable_types.default(data, columns)
-}
-
-#' @export
-variable_types.DataTable <- function(data, columns = NULL) {
-  variable_types.default(data, columns)
-}
-
-#' @export
-variable_types.DFrame <- function(data, columns = NULL) {
-  variable_types.default(data, columns)
-}
-
-#' @export
-variable_types.matrix <- function(data, columns = NULL) {
-  checkmate::assert_character(columns, null.ok = TRUE, any.missing = FALSE)
-
-  res <- if (is.null(columns)) {
-    apply(
-      data,
-      2,
-      function(x) class(x)[1]
-    )
-  } else if (checkmate::test_character(columns, any.missing = FALSE)) {
-    stopifnot(
-      all(
-        columns %in% colnames(data) |
-          vapply(columns, identical, logical(1L), "")
-      )
-    )
-    vapply(
-      columns,
-      function(x) ifelse(x == "", "", class(data[, x])[1]),
-      character(1),
-      USE.NAMES = FALSE
-    )
+  checkmate::assert_multi_class(data, c("data.frame", "DFrame", "matrix"))
```

Comment:
I realized "DFrame" inherits from "DataFrame". Please change to `DataFrame` to include more cases.
<img width="903" alt="Skärmavbild 2024-01-08 kl  11 50 57" src="https://github.com/insightsengineering/teal.slice/assets/6959016/70362d2a-eb69-4d8f-aecd-47f15079f940">


```suggestion
  checkmate::assert_multi_class(data, c("data.frame", "DataFrame", "matrix"))
```

Reply:
The `DataFrame` function produces a `DFrame` object:
```
> DataFrame() %>% class
[1] "DFrame"
attr(,"package")
[1] "S4Vectors"
```

Do we support `SQLDataFrame` and `ParquetDataFrame`? I don't think we depend on packages they are defined in.

Reply:
Right, I see now that 
```
> DataFrame() %>% is("DataFrame")
[1] TRUE
```

Reply:
> DataFrame() %>% class 

This is surprising. However in S4Vectors there is "RectangularData" class which is a parent of `DataFrame` and `DelayedMatrix`. I don't expect any of these objects to be present in MAE@colData but this is a tip for this PR https://github.com/insightsengineering/teal.slice/pull/498